lib/improver/tests/utilities/test_cube_metadata.py

```diff
@@ -192,17 +200,27 @@ class Test_update_coord(IrisTest):
     """Test the update_coord method."""
 
     def test_basic(self):
-        """Test update_coord returns a Cube and updates coord correctly. """
+        """Test update_coord returns a Cube and updates coordinate points and
+        bounds correctly. """
         cube = create_cube_with_threshold()
-        changes = {'points': [2.0], 'bounds': [0.1, 2.0], 'units': 'mm'}
+        changes = {
+            'points': [2.0],
+            'bounds': [0.1, 2.0],
+             }
         result = update_coord(cube, 'threshold', changes)
         self.assertIsInstance(result, Cube)
-        self.assertArrayEqual(result.coord('threshold').points,
-                              np.array([2.0], dtype=np.float32))
-        self.assertArrayEqual(result.coord('threshold').bounds,
-                              np.array([[0.1, 2.0]], dtype=np.float32))
-        self.assertEqual(str(result.coord('threshold').units),
-                         'mm')
+        self.assertArrayAlmostEqual(result.coord('threshold').points,
+                                    np.array([2.0], dtype=np.float32))
+        self.assertArrayAlmostEqual(result.coord('threshold').bounds,
+                                    np.array([[0.1, 2.0]], dtype=np.float32))
+
+    def test_update_units(self):
+        """Test update_coord returns a Cube and updates units correctly. """
+        cube = create_cube_with_threshold()
+        changes = {'units': 'km s-1'}
+        result = update_coord(cube, 'threshold', changes)
+        self.assertIsInstance(result, Cube)
+        self.assertEqual(str(result.coord('threshold').units), 'km s-1')
```

Comment:
Is it worth checking the points in the threshold coord too? Can maybe the word "update" should be "convert"?

Reply:
I've now updated this test as suggested.